packages/flutter/lib/src/widgets/sliver_prototype_item_list.dart

```diff
@@ -0,0 +1,183 @@
+
+  @override
+  void removeChildRenderObject(RenderBox child) {
+    if (renderObject.child == child)
+      renderObject.child = null;
+    else
+      super.removeChildRenderObject(child);
+  }
+
+  @override
+  void visitChildren(ElementVisitor visitor) {
+    if (_prototype != null)
+      visitor(_prototype);
+    super.visitChildren(visitor);
+  }
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    _prototype = updateChild(_prototype, widget.prototypeItem, _prototypeSlot);
+  }
+
+  @override
+  void update(SliverPrototypeExtentList newWidget) {
+    super.update(newWidget);
+    assert(widget == newWidget);
+    _prototype = updateChild(_prototype, widget.prototypeItem, _prototypeSlot);
+  }
+}
+
+class _RenderSliverPrototypeExtentList extends RenderSliverFixedExtentBoxAdaptor {
+  _RenderSliverPrototypeExtentList({
+    @required _SliverPrototypeExtentListElement childManager,
+  }) : super(childManager: childManager);
+
+  RenderBox _child;
+  RenderBox get child => _child;
+  set child(RenderBox value) {
+    if (_child != null)
+      dropChild(_child);
+    _child = value;
+    if (_child != null)
+      adoptChild(_child);
+    markNeedsLayout();
+  }
+
+  @override
+  void performLayout() {
+    child.layout(constraints.asBoxConstraints(), parentUsesSize: true);
+    super.performLayout();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    if (_child != null)
+      _child.attach(owner);
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    if (_child != null)
+      _child.detach();
+  }
+
+  @override
+  void redepthChildren() {
+    if (_child != null)
+      redepthChild(_child);
+    super.redepthChildren();
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    if (_child != null)
+      visitor(_child);
+    super.visitChildren(visitor);
+  }
+
+  @override
+  double get itemExtent {
+    assert(child != null && child.hasSize);
+    return constraints.axis == Axis.vertical ? child.size.height : child.size.width;
+  }
+}
+
+/// A sliver that places its box children in a linear array and constrains them
+/// to have the same extent as a prototype item along the main axis.
+///
+/// [SliverPrototypeExtentList] places its children in a linear array along
+/// the main axis starting at offset zero and without gaps. Each child is
+/// constrained to the same extent as the [prototypeItem] along the main axis
+/// and the [SliverConstraints.crossAxisExtent] along the cross axis.
+///
+/// [SliverPrototypeExtentList] is more efficient than [SliverList] because
+/// [SliverPrototypeExtentList] does not need to layout its children to obtain
+/// their extent along the main axis. It's a little more flexible than
+/// [SliverFixedExtentList] because there's no need to determine the approriate
+/// item extent in pixels.
+///
+/// See also:
+///
+///  * [SliverFixedExtentList], which has a configurable [itemExtent].
```

Comment:
replace "which has a configurable [itemExtent]" with "whose [itemExtent] is configured in pixels" or some such